mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -226,11 +226,14 @@ class PowerIterationClustering private[clustering] (
    */
   private def pic(w: Graph[Double, Double]): PowerIterationClusteringModel = {
     val v = powerIter(w, maxIterations)
-    val assignments = kMeans(v, k).mapPartitions({ iter =>
+    val kMeansModel = kMeans(v, k)
+    val assignments = kMeansModel.mapPartitions({ iter =>
       iter.map { case (id, cluster) =>
         Assignment(id, cluster)
       }
-    }, preservesPartitioning = true)
+    }, preservesPartitioning = true).cache()
+    assignments.count()
+    kMeansModel.unpersist()
```

Comment:
I think we can just avoid having the result of kMeans be persisted to begin with, see below

Reply:
Done.